app/models/miq_provision/state_machine.rb

```diff
@@ -43,6 +43,9 @@ def poll_destination_powered_on_in_provider
 
   def poll_destination_in_vmdb
     update_and_notify_parent(:message => "Validating New #{destination_type}")
+    if dest_name && phase_context.fetch_path(:new_vm_ems_ref, :vm_name).nil?
```

Comment:
@zhitongLBN Not all providers store a hash in `phase_context[:new_vm_ems_ref]` so this logic should be moved to the provider repo.  In this case https://github.com/ManageIQ/manageiq-providers-azure.

Also, wondering under what conditions you find the `vm_name` property is not populated?